File: iOS/ass3/ass3/Data/TouchExerciseRecord.swift
//
//  TouchExerciseRecord.swift
//  ass3
//
//  Created by Thomas Dang on 23/5/2022.
//

import Foundation
import Firebase
import FirebaseFirestoreSwift

public struct TouchExerciseRecord : Codable {
    @DocumentID var documentID: String?
    var startTime: String?
    var endTime: String?
    var exerciseMode: String?
    var goalMode: String?
    var nReps: Int?
    var isCompleted: Bool?
    var buttonPressedList: [ButtonPressedRecord]?
    
    var nButtons: Int?
    var randomOrder: Bool?
    var nextButtonIndication: Bool?
}

struct ButtonPressedRecord : Codable {
    var time: String
    var buttonPressed: Int
}

File: Android/KIT305_Ass2/app/src/main/java/au/edu/utas/dangmt/kit305_ass2/data/TouchExerciseRecord.kt
package au.edu.utas.dangmt.kit305_ass2.data

import java.io.Serializable

data class TouchExerciseRecord (
    var id : String? = null,
    var startTime : String? = null,
    var endTime : String? = null,
    var exerciseMode : String? = null,  // Possible values: "freeplay", "goal"
    var goalMode : String? = null,      // Possible values: "time", "rep"
    var nReps : Int? = null,
    @field:JvmField
    var isCompleted : Boolean? = null,
    var buttonPressedList : MutableList<ButtonPressedRecord>? = null,

    var nButtons : Int? = null,
    var randomOrder : Boolean? = null,
    var nextButtonIndication : Boolean? = null,
    var buttonSize : Float? = null
)

data class ButtonPressedRecord (
    var time : String? = null,
    var buttonPressed : Int? = null
)